docs/v1/api.md

```diff
@@ -0,0 +1,335 @@
+
+## Persistent Volumes
+
+First, this problem space is similar to that of persistent volumes. We model
+the data structures herein after those in persistent volumes primarily because
+we believe that consistency is a virtue. More importantly, the community
+benefits when similar problems have similar solutions.
+
+Specifically, we believe that the following existing types provide a good model
+for the mechanics of how the services offered by a broker can be used in
+Kubernetes:
+
+- `StorageClass`
+- `PersistentVolume`
+- `PersistentVolumeClaim`
+
+## Ingress Controllers
+
+Today, operators choose their own implementation of an ingress controller.
+Kubernetes gives the operator the ability to specify any ingress controller,
+and the complete freedom to implement it themselves.
+
+Similarly, permitting operators to implement or select their own service catalog
+controller would encourage freedom of choice and diversity. The analog to an
+ingress controller for the service catalog system is called a _service catalog
+controller_. Like ingress, these controllers run inside a standard pod. Anyone
+can write and deploy a service catalog controller, with no changes necessary to
+the Kubernetes install.
+
+Also as in the case of ingress controllers, a cluster may contain many
+different service catalog controllers, each of which handles operations for a
+designated subset of the applicable resource type. For example, a cluster may
+contain a service catalog controller to handle AWS RDS databases, and a
+completely separate service catalog controller to handle on-cluster Redis
+instances.
+
+As we'll detail below, each service catalog controller implementation must
+conform to an implementation spec, but that spec allows for a heterogenous set
+of controllers in a cluster. We believe this allowance encourages flexibility
+in a cluster's service catalog.
+
+# Kubernetes Resources
+
+Given these inspirations, the following resources will be added to the
+Kubernetes API:
+
+- `ServiceClass`
+  - Describes a _kind_ of service. This resource is semantically similar to how
+    a `StorageClass` defines a _kind_ of persistent volume, but we expect that,
+    in practice, the kinds of services represented here will be more coarse-
+    grained than the kinds of volumes represented in `StorageClass`es
+  - References a service ID and plan ID, which is managed and understood by a
+    service catalog controller (see below)
+  - Created by an operator or tool
+- `ServiceInstance`
+  - An instance of a `ServiceClass` -- an in-cluster representation of a
+    provisioned service
+  - Can be statically provisioned by an operator, to enable multiple consumers
+    to bind to it (enables multi-tenancy)
+  - Can be dynamically provisioned and bound to by a consumer, to guarantee an
+    app’s exclusive use of the resource (guarantees single-tenancy)
+- `ServiceClassClaim`
+  - Created by a consumer to provision an instance of a `ServiceClass`, then
```

Comment:
A "Claim" is used to create the service instance AND the binding? I would think that creating the ServiceInstance would do the first part - otherwise how do we bind the same instance to two apps?